### PR TITLE
support domain specific conf and use merge-config

### PR DESCRIPTION
**Pull request by dlozlla**
It also includes server-terminate

### CODE REVIEW

File: lib/services/ldap-provider.js [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
-  maxConnections: config.ldap.maxConnections,
+  url: ldapConfig.url,
+  maxConnections: ldapConfig.maxConnections
 });
 
 var validate = function(email, cb) {
+  //obtain email domain
+  var parts = email.split('@');
+  var emailDomain = parts[1];
+  var domain = (ldapConfig.domains[emailDomain] === undefined) ? 'default' : emailDomain;
```
Comment (ealogar) [inline lib/services/ldap-provider.js]: var domain = emailDomain in ldapConfig ? 'default' : emailDomain

File: lib/services/ldap-provider.js [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
+  var filterValue = null;
+
+  //if there is regex configured for that domain, apply it, otherwise just take email as received
+  if (ldapConfig.domains[domain].filterFieldRegEx) {
```
Comment (ealogar) [inline lib/services/ldap-provider.js]: I would avoid this if if we just force filterFieldRegEx to be always defined. You can use jsonschema...

File: lib/services/ldap-provider.js (hunk 1 above)
Comment (dlozlla) [inline lib/services/ldap-provider.js]: This is python, right?

**Comment by ealogar** [comment]
LGTM

**Pull request merged**